src/EditorFeatures/Test2/IntelliSense/CSharpCompletionCommandHandlerTests.vb

```diff
@@ -12324,5 +12328,95 @@ public static class Ext
                 Assert.NotEmpty(state.GetCompletionItems())
             End Using
         End Sub
+
+        <WpfTheory>
+        <InlineData("task$$", True)>
+        <InlineData("task$$", False)>
+        <InlineData("class C { void M() { c$$ } }", True)>
+        <InlineData("class C { void M() { c$$ } }", False)>
+        <InlineData("class C { void M() { System.Threading.CancellationToken $$ } }", True)>
+        <InlineData("class C { void M() { System.Threading.CancellationToken $$ } }", False)>
+        <InlineData("class C { void M(string x) { x.$$ } }", True)>
+        <InlineData("class C { void M(string x) { x.$$ } }", False)>
+        <InlineData("class C
+        {
+            override $$
+        }", True)>
+        <InlineData("class C
+        {
+            override $$
+        }", False)>
+        <WorkItem("https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1884712")>
+        Public Async Function TestCalculatingCompletionDoNotRunSourceGenerator(code As String, hasCompilationAvailable As Boolean) As Task
```

Comment:
This is not an exhaustive list of scenarios, but should cover the most providers we have internally